gapir/cc/interpreter.cpp

```diff
@@ -411,99 +749,135 @@ Interpreter::Result Interpreter::strcpy(uint32_t opcode) {
   for (; i < count; i++) {
     target[i] = 0;
   }
-  return mStack.isValid() ? SUCCESS : ERROR;
+  return SUCCESS;
 }
 
 Interpreter::Result Interpreter::extend(uint32_t opcode) {
   uint32_t data = extract26bitData(opcode);
-  auto type = mStack.getTopType();
-  auto value = mStack.popBaseValue();
-  switch (type) {
-    // Masking out the mantissa end extending it with the new bits for floating
-    // point types
-    case BaseType::Float: {
-      value |= (data & 0x007fffffULL);
-      break;
-    }
-    case BaseType::Double: {
-      uint64_t exponent = value & 0xfff0000000000000ULL;
-      value <<= 26;
-      value |= data;
-      value &= 0x000fffffffffffffULL;
-      value |= exponent;
-      break;
-    }
-    // Extending the value with 26 new LSB
-    default: {
-      value = (value << 26) | data;
-      break;
+  const std::type_index& type = mStack.typeAtTop();
```

Comment:
Are we compiling with RTTI turned on?
I would hope that we are not .

Reply:
Why would you hope that we are not?

Reply:
RTTI has historically been poorly supported on Android.
Also it has a non-trivial performance cost for all objects regardless of if RTTI is used in that object or not.

The second reason is that it is good practice for the developers of this tool to apply similar constraints as our users have. If only to have more practice with how our users work. (Most game developers disable RTTI & Exceptions)

Reply:
Like all things it, the appropriateness of RTTI depends on when and where you use it. Much of the usage here is compile time evaluated and has zero overhead. I changed the code because vtune was telling me that we were spending more time in our own RTTI replacement than we would spend using C++'s built in type system. Slower and non-standard loses to faster and standard in my books.